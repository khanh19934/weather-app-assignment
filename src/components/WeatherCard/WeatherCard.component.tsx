import React from 'react';
import { format } from 'date-fns';

interface IWeather {
  icon: string;
  description: string;
}

interface IProps {
  dt: number;
  weather: IWeather[];
  temp: {
    min: number;
    max: number;
  };
}

const WeatherCard: React.FC<IProps> = ({ dt, weather, temp }: IProps) => {
  return (
    <div className='forecast-day' data-testid='weatherCard'>
      <img
        src={`https://openweathermap.org/img/wn/${weather[0].icon || '02d'}@2x.png`}
        alt='weather'
        className='weather-icon-card'
      />
      <span>{format(dt * 1000, 'ccc')}</span>
      <span>
        {Math.round(temp.min)} ° - {Math.round(temp.max)} °
      </span>
    </div>
  );
};

export default WeatherCard;
