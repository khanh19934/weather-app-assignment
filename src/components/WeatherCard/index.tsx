export { default } from './WeatherCard.component';
