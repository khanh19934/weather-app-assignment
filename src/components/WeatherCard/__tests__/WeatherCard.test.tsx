import React from 'react';
import renderer from 'react-test-renderer';
import WeatherCard from '../WeatherCard.component';

it('render correctly', () => {
  const tree = renderer
    .create(
      <WeatherCard dt={12} temp={{ min: 1, max: 10 }} weather={[{ description: '', icon: '' }]} />,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
