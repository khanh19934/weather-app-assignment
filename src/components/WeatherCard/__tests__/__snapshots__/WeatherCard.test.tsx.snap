// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<div
  className="forecast-day"
  data-testid="weatherCard"
>
  <img
    alt="weather"
    className="weather-icon-card"
    src="https://openweathermap.org/img/wn/02d@2x.png"
  />
  <span>
    Thu
  </span>
  <span>
    1
     ° - 
    10
     °
  </span>
</div>
`;
